File: src/scala/org/codehaus/testdox/intellij/TestDoxClass.scala
package org.codehaus.testdox.intellij

import com.intellij.openapi.vfs.VirtualFile;

import org.codehaus.testdox.intellij.panel.TestDoxModel;

class TestDoxClass(override val file: VirtualFile,
                   override val className: String,
                   private val isTheTestedClass: Boolean,
                   override val testClass: TestClass,
                   override val testedClass: TestClass,
                   override val testMethods: Array[TestMethod]) extends TestDoxFile(file, className, testClass, testedClass, testMethods) {

  override val isTestedClass = isTheTestedClass

  override val canBeUnitTested = true

  override def canNavigateToTestClass = testClass != null && testClass.isTestClass

  override def canNavigateToTestedClass = testedClass != null && !testedClass.isTestClass

  def updateModel(model: TestDoxModel) {
    model.setTestDoxForClass(this)
  }
}

object TestDoxClass {
  val NO_DOX_ELEMENT: TestElement = new AbstractTestElement() {
    val displayString = "<font color=\"red\">No tests found for current class!</font>"
    val icon = IconHelper.getIcon(IconHelper.NO_TESTS_ICON)
  }
}

File: src/java/org/codehaus/testdox/intellij/TestDoxClass.java
package org.codehaus.testdox.intellij;

import javax.swing.Icon;

import com.intellij.openapi.vfs.VirtualFile;

import org.codehaus.testdox.intellij.panel.TestDoxModel;

public class TestDoxClass extends TestDoxFile {

    public static final TestElement NO_DOX_ELEMENT = new AbstractTestElement() {
        public String displayString() {
            return "<font color=\"red\">No tests found for current class!</font>";
        }

        public Icon icon() {
            return IconHelper.getIcon(IconHelper.NO_TESTS_ICON);
        }
    };
    private boolean isTestedClass;

    public TestDoxClass(VirtualFile file, String className, boolean isTestedClass, TestClass testClass,
                        TestClass testedClass, TestMethod[] testMethods) {
        super(file, className, testClass, testedClass, testMethods);
        this.isTestedClass = isTestedClass;
    }

    public void updateModel(TestDoxModel model) {
        model.setTestDoxForClass(this);
    }

    public boolean isTestedClass() {
        return isTestedClass;
    }

    public boolean canBeUnitTested() {
        return true;
    }

    public boolean canNavigateToTestClass() {
        return (testClass() != null) && (testClass().isTestClass());
    }

    public boolean canNavigateToTestedClass() {
        return (testedClass() != null) && (!testedClass().isTestClass());
    }
}
